openshift-kube-apiserver/handlers/metricsforwarding/metricforwarding.go

```diff
@@ -0,0 +1,102 @@
+// used to provide metric access via check-endpoints in kube-apiserver static pods
+package metricsforwarding
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+type metricForwarding struct {
+	client *http.Client
+}
+
+func newMetricForwardingHandler() (*metricForwarding, error) {
+	clientConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: "/etc/kubernetes/static-pod-certs/configmaps/control-plane-node-kubeconfig/kubeconfig"}, nil).
+		ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	// these are required to create the RESTClient. We want the REST client to get the correct authn/authz behavior
+	clientConfig.GroupVersion = &schema.GroupVersion{Version: "if-this-is-used-find-the-bug"}
```

Comment:
using a rest client with all boilerplate seems odd. Why not simple tcp client?

Reply:
> the correct authn/authz behavior

This means what? The client cert?

Reply:
> This means what? The client cert?

client-cert, error handling, error parsing for returns, default decoding that recognizes api.Status, logging verbosity, forwarding and retry logic.